destination.go

```diff
@@ -1,77 +1,203 @@
 }
 
 func (d *Destination) Parameters() config.Parameters {
-	// Parameters is a map of named Parameters that describe how to configure
-	// the Destination. Parameters can be generated from DestinationConfig with
-	// paramgen.
 	return d.config.Parameters()
 }
 
 func (d *Destination) Configure(ctx context.Context, cfg config.Config) error {
-	// Configure is the first function to be called in a connector. It provides
-	// the connector with the configuration that can be validated and stored.
-	// In case the configuration is not valid it should return an error.
-	// Testing if your connector can reach the configured data source should be
-	// done in Open, not in Configure.
-	// The SDK will validate the configuration and populate default values
-	// before calling Configure. If you need to do more complex validations you
-	// can do them manually here.
-
-	sdk.Logger(ctx).Info().Msg("Configuring Destination...")
+	sdk.Logger(ctx).Info().Msg("configuring destination...")
 	err := sdk.Util.ParseConfig(ctx, cfg, &d.config, NewDestination().Parameters())
 	if err != nil {
 		return fmt.Errorf("invalid config: %w", err)
 	}
 	return nil
 }
 
-func (d *Destination) Open(_ context.Context) error {
-	// Open is called after Configure to signal the plugin it can prepare to
-	// start writing records. If needed, the plugin should open connections in
-	// this function.
+func (d *Destination) Open(ctx context.Context) error {
+	d.client = openai.NewClient(d.config.APIKey)
+
+	// check that the passed api key is valid, so that we ensure that the api
+	// calls have no auth errors.
+
+	if _, err := d.client.ListModels(ctx); err != nil {
+		return fmt.Errorf("failed to validate api key: %w", err)
+	}
+
+	sdk.Logger(ctx).Info().Msg("api key is valid")
+
+	_, err := d.client.RetrieveVectorStore(ctx, d.config.VectorStoreID)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve vector store %s: %w", d.config.VectorStoreID, err)
+	}
+
+	return nil
+}
+
+func (d *Destination) Write(ctx context.Context, recs []opencdc.Record) (int, error) {
+	for i, rec := range recs {
+		switch rec.Operation {
+		case opencdc.OperationCreate, opencdc.OperationSnapshot:
+			// We want creates and snapshots to not leave duplicated files, so we
+			// interpret them as an upsert
+			if err := d.upsertFile(ctx, rec); err != nil {
+				return i, err
+			}
+		case opencdc.OperationUpdate:
+			if err := d.upsertFile(ctx, rec); err != nil {
+				return i, err
+			}
+		case opencdc.OperationDelete:
+			listedFiles, err := d.client.ListFiles(ctx)
+			if err != nil {
+				return i, fmt.Errorf("failed to list files: %w", err)
+			}
+
+			if err := d.deleteFile(ctx, rec, listedFiles); err != nil {
+				return i, err
+			}
+		}
+	}
+
+	return len(recs), nil
+}
+
+func (d *Destination) createFile(ctx context.Context, rec opencdc.Record) error {
+	filename := string(rec.Key.Bytes())
+	filebs := rec.Payload.After.Bytes()
+	f, err := d.client.CreateFileBytes(ctx, openai.FileBytesRequest{
```

Comment:
I mean, we don't check the bytes length aswell when writing to s3:
https://github.com/ConduitIO/conduit-connector-s3/blob/bf19253d1141ef5501fa10401749de23a22241fb/destination/writer/s3.go#L86-L111

I mean we could give a Warn log if that happens, but I would not prevent it. I feel like it would make the connector a bit more difficult to undersand.